### ISSUE TITLE
Event View too small for no particular reason

### ISSUE THREAD

**Issue by theprojectguy** [richardtop/CalendarKit#260]
### New Issue Checklist

- [x] Use the latest version of CalendarKit
- [x] I've read the [Contribution Guidelines](https://github.com/richardtop/CalendarKit/blob/master/CONTRIBUTING.md)
- [x] I've searched for [existing GitHub issues](https://github.com/richardtop/CalendarKit/issues)

### Issue Description
Before the description, I wanted to say thank you for this amazing library !
So, I have several (a lot of) activities in a day, and sometimes the event view height is too small for no reason because there is a lot space to fill.
I don't really know which part of my code to show you, but I use CalendarKit as mentioned in the tutorials.
This is a screenshot of my problem:

<img width="331" alt="Screenshot 2020-05-04 at 07 02 15" src="https://user-images.githubusercontent.com/40181807/80938043-df2d6280-8dd7-11ea-89c8-b50d30a289f3.png">

##### Code I'm using with CalendarKit
```swift
 override func eventsForDate(_ date: Date) -> [EventDescriptor] {
    var events = [Event]()
    let currentDayActivities = self.calendarManager?.getDayActivitiesByDate(date, MonthsActivities: self.allActivities)

    if let activitiesData = currentDayActivities {
      for activity in activitiesData {
      let event = Event()
      let duration = activity.activityDuration
      let datePeriod = TimePeriod(beginning: activity.start!,
                                            chunk: TimeChunk.dateComponents(minutes: duration))
      event.startDate = datePeriod.beginning!
      event.endDate = datePeriod.end!
      if let currentColor = self.calendarManager?.getActivityColor(activity) {
                   event.color = currentColor
      }
      let info = [activity.activityTitle, ""]
      event.text = info.reduce("", {$0 + $1 + "\n"})
      events.append(event)
      }
}
```

#### Result I am trying to achieve
Sometimes, Event View activities are well divided inside the Calendar so here is a good result for me:

<img width="331" alt="Screenshot 2020-05-04 at 07 08 43" src="https://user-images.githubusercontent.com/40181807/80937989-98d80380-8dd7-11ea-9470-7f9b2089e8a5.png">

So, do you think it's a bug ?
And do you have any suggestions to solve it temporarily ?

Thank you in advance !

**Comment by theprojectguy** [comment]
MB
I just saw issue #113 just after the publication of this one. So mine is a duplicated issue (Sorry !).
But do you have any recommendation or solution to add ?

**Comment by richardtop** [comment]
Hi, thanks for raising this issue. As you've discovered, it's a known bug in the layout algorithm.

In short, it's due to the fact that the algorithm doesn't try to layout the events on top of each other. Rather, it splits the events into multiple columns. If there are multiple long overlapping events, this results in a lot of free space not used. 

Usually that means, that the actual events data provided to the calendar kit is too much and wouldn't fit in well anyway.

However, as you've shown in your first example, that's definitely not the case here and the events could be laid out in a better way.

The layout algorithm in question is in the [TimelineView.swift#L380-L438](https://github.com/richardtop/CalendarKit/blob/aa1cab4f54cc624d9eec3a4750c6ffa8f5cf344c/Source/Timeline/TimelineView.swift#L380-L438) and I hope, it's pretty clear, how it works.

Feel free to improve it and propose a pull request.

Alternatively, you can take a look at the `TimelineStyle.eventsWillOverlap` property, as enabling / disabling it might help in your case.

**Comment by richardtop** [comment]
Closed in favor of #113
